### ISSUE TITLE
Infinitely connecting when one connection for the same torrent file has been closed

### ISSUE THREAD

**Issue by adler-amorette** [popcorntime/popcorntime#3041]
#### Operating System Version:
LMDE 6

#### Popcorn Time Version:
Popcorn-Time-0.5.1-amd64-0.44.5.deb

#### Download url:
github releases

#### Expected Behaviour:
Open the torrent file as if the app has just been opened.

#### Actual Behaviour:
I get the endless "Connecting" page.

#### Steps to reproduce the behaviour:
1. Open the app
2. Open a torrent file and click on a video file to play.
3. Cancel before the player has opened, or close the player at any point. (I use the built-in player)
4. Drag over or open from "Saved torrents" the same torrent file

**Comment by merseyV** [comment]
Might as well add my tuppence worth ... notwithstanding I'm a Mac user!!

I'm using the 0.86.0 runtime version .. but I've had similar with 0.44.5 in the past: Open/close/re-open same torrent (built-in player) hangs sometimes. Requires a 2nd attempt to reopen connection ... consistently. Previously I'd do a clean reboot (probably overkill) and it would clear things up.

Two things I've stumbled across (which may well be unrelated to OP's problem?)
* If it happens to a "stored torrent" and a 2nd attempt (to open) doesn't work, I'd assume it's corrupted? Delete the file and restore copy  .. works as normal, again.
* This one I can't even guess at: The consistent 2 attempts to open has some link/involvement with customising my 'cache directory' location. Doesn't happen straight away .. but once it starts playing up, it's consistent. If I delete the "custom entry" in settings.db (return to default location) it works "normally" again.

Sounds bizarre .. but there you go!! 😃

**Comment by merseyV** [comment]
Might as well add my tuppence worth ... notwithstanding I'm a Mac user!!

I'm using the 0.86.0 runtime version .. but I've had similar with 0.44.5 in the past: Opening torrents, used to hang (first-time), sometimes. Required a 2nd attempt to reopen connection ... persistently. Previously I'd just do a clean reboot (probably overkill) and it would clear things up.

Two things I've stumbled across (which may well be nothing to do with OP's problem?)
* If it happens to a "stored torrent" and a 2nd attempt (to open it) doesn't work, I'd assume it's a corrupted link? Delete the file and restore copy  .. worked as normal then..
* This one I can't even guess at: When it needed 2 attempts to open a connection (torrent link), every time ... If I deleted the "custom entry" in settings.db and rebooted (returning it to the default location) .. it worked "normally" again. Why a link to my customised 'cache directory' location? - I haven't a clue! 

Totally bizarre, but there you go? ... It's Voodoo!!😃
